R/coh_effectiveness.R

```diff
@@ -0,0 +1,190 @@
+#' vaccineff data.
+#' @param at Number of days at which VE is estimated from the beginning of the
+#' follow-up period. If NULL, effectiveness is estimated at the end of the
+#' study, assigned as `end_cohort`. Default is NULL.
+#' @return Object of the class `effectiveness`: a list with results from the
+#' estimation of VE.
+#' `ve`: `data.frame` with VE(CI95%)
+#' `cox_model`: `survival` object with Cox model results
+#' `kaplan_meier`: `survival` object with Kaplan-Meier estimator
+#' @examples
+#' # Load example data
+#' data("cohortdata")
+#'
+#' # Create `vaccineff_data`
+#' vaccineff_data <- make_vaccineff_data(data_set = cohortdata,
+#'   outcome_date_col = "death_date",
+#'   censoring_date_col = "death_other_causes",
+#'   vacc_date_col = "vaccine_date_2",
+#'   vaccinated_status = "v",
+#'   unvaccinated_status = "u",
+#'   immunization_delay = 15,
+#'   end_cohort = as.Date("2044-12-31"),
+#'   match = TRUE,
+#'   exact = c("age", "sex"),
+#'   nearest = NULL
+#' )
+#'
+#' # Estimate the Vaccine Effectiveness (VE)
+#' ve <- effectiveness(vaccineff_data, 90)
+#'
+#' # Print summary of VE
+#' summary(ve)
+#'
+#' # Generate loglog plot to check proportional hazards
+#' plot(ve, type = "loglog")
+#'
+#' # Generate Survival plot
+#' plot(ve, type = "surv", percentage = FALSE, cumulative = FALSE)
+#' @export
+
+effectiveness <- function(vaccineff_data,
+                          at = NULL) {
+
+  stopifnot("Input must be an object of class 'vaccineff_data'" =
+      checkmate::test_class(vaccineff_data, "vaccineff_data")
+  )
+
+  # Check numeric argument for at
+  checkmate::test_integerish(at, lower = 0, null.ok = TRUE)
+
+  if (!is.null(vaccineff_data$matching)) {
+    data_set <- vaccineff_data$matching$match
+  } else {
+    data_set <- vaccineff_data$cohort_data
+  }
+
+  tags <- linelist::tags(data_set)
+
+  if (!is.null(at)) {
+    data_set <- truncate_time_to_event(
+      data_set = data_set,
+      outcome_date_col = tags$outcome_date_col,
+      end_cohort = vaccineff_data$end_cohort,
+      at = at
+    )
+  }
+
+  eff_obj <- coh_eff_hr(
+    data_set = data_set,
+    outcome_status_col = tags$outcome_status_col,
+    time_to_event_col = tags$time_to_event_col,
+    vacc_status_col = tags$vacc_status_col,
+    vaccinated_status = vaccineff_data$vaccinated_status,
+    unvaccinated_status = vaccineff_data$unvaccinated_status,
+    start_cohort = vaccineff_data$start_cohort,
+    end_cohort = vaccineff_data$end_cohort
+  )
+  # Estimation at
+  eff_obj$at <- at
+  # effectiveness object
+  class(eff_obj) <- "effectiveness"
+
+  # return
+  return(eff_obj)
+}
+
+#' @title Summarize VE Results
+#'
+#' @description Summarizes the results of `effectiveness`.
+#'
+#' @param object Object of the class `effectiveness`.
+#' @param ... Additional arguments passed to other functions.
+#' @return Summary of the results from effectiveness.
+#' @export
+
+summary.effectiveness <- function(object, ...) {
```

Comment:
`summary.effectiveness()` doesn't seem to be returning anything, and when running the function the output is a string of one of the print messages. It's fine to prints stuff to console when the user calls a function such as `summary()` but for it to behave more similarly to other `summary()` methods I'd recommend returning what is currently printed and then you can also print it to console as well. I'd also recommend using `message()` rather than `cat()` in functions where their intended purpose is not to print (e.g. `print()`) as messages can be more easily silenced by users.

Reply:
I agree with this. I will refactorize all summary methods to have the same behaviour. See issue #89